src/middlewared/middlewared/plugins/catalogs_linux/update.py

```diff
@@ -265,9 +265,12 @@ async def do_create(self, job, data):
         if not data.pop('force'):
             job.set_progress(40, f'Validating {data["label"]!r} catalog')
             # We will validate the catalog now to ensure it's valid wrt contents / format
+            k8s_dataset = (await self.middleware.call('kubernetes.config'))['dataset']
             path = os.path.join(
-                TMP_IX_APPS_DIR, 'validate_catalogs', convert_repository_to_path(data['repository'], data['branch'])
+                '/mnt', k8s_dataset, 'catalogs/validate_catalogs',
+                convert_repository_to_path(data['repository'], data['branch'])
             )
+            await self.middleware.run_in_thread(shutil.rmtree, path, ignore_errors=True)
```

Comment:
Why are we doing this? This gets run in the `finally:` block down below?

Reply:
This is to ensure that the directory does not exist before we try to clone into it

Reply:
In ideal situations, almost always the directory won't be there and this is just a failsafe

Reply:
This still gets consumed for official catalog when we sync catalog. Basically we still allow syncing official catalog to tmpfs to allow users to view available apps if they don't have apps configured (https://github.com/truenas/middleware/pull/12839).